Pia 2838 help screen updates

chriskryn opened the pull request:
- view controllers updates
- data sources
- file structures
- design

zladzeyka commented on BankSDK/GiniBankSDKExample/GiniBankSDKExample/ComponentAPI/ComponentAPICoordinator.swift:
```diff
@@ -226,10 +226,13 @@ extension ComponentAPICoordinator {
         if pages.type == .image {
-            let imageAnalysisNoResultsViewController = ImageAnalysisNoResultsViewController()
-            imageAnalysisNoResultsViewController.didTapBottomButton = { [unowned self] in
+            let imageAnalysisNoResultsViewController = HelpTipsViewController(giniConfiguration: giniBankConfiguration.captureConfiguration())
```
@chriskryn  I'm wondering why we need this change. Maybe it is better to revert it and comment out broken lines until the new NoResultsViewController will be implemented. What do you think?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpBaseDataSource.swift:
```diff
@@ -0,0 +1,58 @@
+    
+    @objc(tableView:willDisplayCell:forRowAtIndexPath:) public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        if items.count == 1 {
+          cell.round(corners: [.bottomLeft, .bottomRight, .topLeft, .topRight], withRadius: 8)
```
what about adding a radius corner as a constant?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpMenu/HelpMenuDataSource.swift:
```diff
@@ -0,0 +1,56 @@
+    ) {
+        super.init(configuration: configuration)
+        self.items.append(contentsOf: defaultItems)
+        self.items.append(contentsOf: giniConfiguration.customMenuItems)
```
Should I pick custom items from configuration or giniConfiguration?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpImportViewController.swift:
```diff
@@ -0,0 +1,122 @@
+        self.title = "How to import"
+        view.addSubview(tableView)
+        view.backgroundColor = UIColor.from(giniColor: giniConfiguration.helpScreenBackgroundColor)
+        edgesForExtendedLayout = []
```
@chriskryn you can remove line 52 - you're doing it already in 45

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpImportViewController.swift:
```diff
@@ -0,0 +1,122 @@
+        cell.headerLabel.backgroundColor = UIColor.clear
+        cell.descriptionLabel.backgroundColor = UIColor.clear
+        cell.contentView.backgroundColor = UIColor.clear
+    }
```
The empty line is missing 
![image](https://user-images.githubusercontent.com/3911816/182868164-6b19b35f-7f78-47a3-a034-ec19a2178e96.png)

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpImportViewController.swift:
```diff
@@ -0,0 +1,122 @@
+            configureCell(cell: cell, indexPath: indexPath)
+            return cell
+        }
+        fatalError()
```
@chriskryn would you mind adding a message for this fatal error 
![image](https://user-images.githubusercontent.com/3911816/182868746-4200b120-6c9b-4697-b8c2-026cdfcbec65.png)

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpMenuViewController.swift:
```diff
@@ -0,0 +1,103 @@
+ 
+ - note: Component API only.
+ */
+
```
@chriskryn could you please add TODO - REMOVE Componen API because we will remove all related Component API code in the nearest future?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpMenuViewController.swift:
```diff
@@ -0,0 +1,103 @@
+    private (set) var dataSource: HelpMenuDataSource
+    private let giniConfiguration: GiniConfiguration
+    private let tableRowHeight: CGFloat = 44
+    
```
Extra empty line :)

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator+Analysis.swift:
```diff
@@ -23,8 +23,8 @@ extension GiniScreenAPICoordinator {
 extension GiniScreenAPICoordinator {
-    func createImageAnalysisNoResultsScreen() -> ImageAnalysisNoResultsViewController {
-        let imageAnalysisNoResultsViewController: ImageAnalysisNoResultsViewController
+    func createImageAnalysisNoResultsScreen() -> HelpTipsViewController {
```
We need to agree, on whether should we keep the old Help screen for ImageAnalysisNoResultsViewController or we add some TODOs about the NoResults screen.

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/ImageNoResultsViewControllerTests.swift:
```diff
@@ -11,8 +11,8 @@ import XCTest
     let viewControllerTitle = "Title"
-    lazy var viewController: ImageAnalysisNoResultsViewController =
-        ImageAnalysisNoResultsViewController(title: self.viewControllerTitle)
+    lazy var viewController: HelpTipsViewController =
```
:) TODO and comment it our or use old NoResultsScreen?

zladzeyka commented on HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/ComponentAPICoordinator.swift:
```diff
@@ -379,7 +379,7 @@ extension ComponentAPICoordinator: UINavigationControllerDelegate {
             (fromVC is ReviewViewController ||
              fromVC is AnalysisViewController ||
-             fromVC is ImageAnalysisNoResultsViewController) {
+             fromVC is HelpTipsViewController) {
```
Please, 🙏 revert this change, we don't touch HealthSDK now ;)

chriskryn replied:
@zladzeyka ok, I will revert this change and old NoResults screen too

zladzeyka replied:
@chriskryn thank you 😊 🙏

zladzeyka reviewed (changes requested):
@chriskryn Great Job 👍 👏 
![image](https://user-images.githubusercontent.com/3911816/182874349-1da9dfe4-b878-47a0-bec9-3d074106de60.png)

I've found only small things: empty lines, the only 1 line of duplication
and no result screen case - I would recommend keeping the old No Result if it's possible, it will make it much easier to reimplement it ( there are a few comments about it).
An image named XXXmultiple.png should be a pdf.

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpBaseDataSource.swift:
```diff
@@ -0,0 +1,58 @@
+        return 1
+    }
+    
+    @objc(tableView:willDisplayCell:forRowAtIndexPath:) public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
```
Is it a new style how to add @objc? I've never seen it.

chriskryn replied:
Good catch! hmm ... It was automatically added by Xcode after some complains. I need to check it maybe it's no longer necessary

chriskryn replied:
It seems so, but it's no longer needed it seems redundant. I will remove it

zladzeyka commented on BankSDK/GiniBankSDKExample/GiniBankSDKExample/ComponentAPI/ComponentAPICoordinator.swift:
```diff
@@ -472,7 +472,7 @@ extension ComponentAPICoordinator: UINavigationControllerDelegate {
                 fromVC is AnalysisViewController ||
-                fromVC is ImageAnalysisNoResultsViewController) {
+                fromVC is HelpTipsViewController) {
             // When going directly from the analysis or from the single page review screen to the camera the pages
```
fromVC is ImageAnalysisNoResultsViewController) {

we need to use ImageAnalysisNoResultsViewController as well

chriskryn replied:
ok, I missed it.

zladzeyka reviewed (changes requested):
Great, only one tiny thing in https://github.com/gini/gini-mobile-ios/pull/122/files#diff-a17b4714b1a188bc781fb173bd0986867bd92257a9b25eada5febf52b0a1b56a

zladzeyka reviewed (approved):
Looks good!
![image](https://user-images.githubusercontent.com/3911816/183058098-bccd05aa-e57d-4826-835f-8ddb8ce2d6ba.png)

chriskryn commented:
@zladzeyka  I have applied the colors and texts changes, could you please have a look again at this ? I will merge it if it looks fine

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpFormatsDataSource.swift:
```diff
@@ -0,0 +1,155 @@
+    
+    lazy var sections: [HelpFormatsCollectionSection] = {
+        var sections: [HelpFormatsCollectionSection] =  [
+            (NSLocalizedString(
```
For all localizable resources, we need to use `NSLocalizedStringPreferredFormat` from GiniCaptureUtils. That will give the possibility to override localizable stings from the app bundle :).

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpFormatsDataSource.swift:
```diff
@@ -0,0 +1,155 @@
+    
+    func tableView(
+        _ tableView: UITableView,
+        cellForRowAt indexPath: IndexPath
```
Could you please apply the formatting here?

zladzeyka reviewed (changes requested):
Very close :)

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpFormatsDataSource.swift:
```diff
@@ -0,0 +1,147 @@
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
```
@chriskryn Could you please add a new line here?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpFormatsDataSource.swift:
```diff
@@ -0,0 +1,147 @@
+        return UITableView.automaticDimension
+    }
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        
```
@chriskryn what do you think should we remove this empty line here?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpMenu/HelpMenuDataSource.swift:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public override func configureCell(cell: HelpMenuCell, indexPath: IndexPath) {
+
```
And remove empty line here  :)

zladzeyka replied:
It is still there 🙈

zladzeyka replied:
line 41

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/DataSources/HelpTipsDataSource.swift:
```diff
@@ -0,0 +1,65 @@
+    }
+    
+    public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
```
I think we can disable selection for UITableView like:
tableView.allowsSelection = false as well. Can we try it?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/ViewControllers/HelpTipsViewController.swift:
```diff
@@ -0,0 +1,77 @@
+    public func configureMainView() {
+        view.addSubview(tableView)
+        view.backgroundColor = UIColorPreferred(named: "systemGray06")
+        edgesForExtendedLayout = []
```
it's already in line 48

zladzeyka reviewed (changes requested):
Thank you @chriskryn 
Almost there, just a few empty lines manipulation and 2 comments left
![image](https://user-images.githubusercontent.com/3911816/183960634-81b27cb4-f4ff-4b4c-b6c0-9b68add61653.png)

zladzeyka commented:
I've tested as well:
My findings:
- Icons to be switched
![Untitled](https://user-images.githubusercontent.com/3911816/183966248-70b9c20c-4942-4341-b2d4-d13f4fc207f5.png)

- Bouncing should be disabled for `Tips for photos`.

- Section header is too big:
 ![Simulator Screen Shot - iPhone 13 - 2022-08-10 at 18 39 47](https://user-images.githubusercontent.com/3911816/183966663-04bf55f9-10b8-4290-9e84-c7a02633898f.png)
 
- Some color for the cell's background is missing. 
![Simulator Screen Shot - iPhone 8 Plus - 2022-08-10 at 18 33 48](https://user-images.githubusercontent.com/3911816/183967397-297de4ed-d3ec-421c-82e7-ad9a08401225.png) Please see light mode below and [design](https://www.figma.com/file/m4OtCvxWxPMeB0Z5k4Aeow/Gini-Photo-Payment-%2F-iOS?node-id=750%3A9151)
![Simulator Screen Shot - iPhone 8 Plus - 2022-08-10 at 18 26 50](https://user-images.githubusercontent.com/3911816/183967985-e7e1691b-2eac-4e5e-ac98-1db11fb64a93.png)

chriskryn commented:
Good catch :+1:. Ok, I have fixed it. The color values again didn't correspond to the color templates. I have added new color into assets as helpBackground. I think we probably should think about naming these kind of new colors.

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/Open with tutorial/OpenWithTutorialCollectionCell.swift:
```diff
@@ -9,130 +9,131 @@
-        stepIndicatorCircle.layer.borderColor = UIColor.from(giniColor: GiniConfiguration.shared.indicatorCircleColor).cgColor
+
+        stepIndicatorCircle.layer.borderColor = UIColor.from(
+            giniColor: GiniConfiguration.shared.indicatorCircleColor).cgColor
```
Can we use here directly UIColorPreferred(named: "indicatorCircleColor") ?

zladzeyka commented on CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Help/Open with tutorial/OpenWithTutorialCollectionCell.swift:
```diff
@@ -9,130 +9,131 @@
-        
+
         if #available(iOS 13.0, *) {
             backgroundColor = .systemBackground
```
@chriskryn  could you please add this to our palette as well
and use UIColorPreferred😉

zladzeyka reviewed (approved):
![image](https://user-images.githubusercontent.com/3911816/184120481-320261af-d9a3-4207-8119-f8c46c8d3099.png)
Great success!
